### PR TITLE
Guarin lig 3072 cleanup readme

### PR DESCRIPTION
**Pull request by guarin**
## Changes

* Update readme
* Add `pylint` to dev dependencies (it was missing)
* Add `test-fast` to makefile for consistency
* Remove linting instructions from readme as they are not enforced in CI
* Add PyPI, downloads, and codestyle badges
* Add link to documentation for all models
* Rewrite code examples using transforms

## How was it tested?

* Tested code example manually

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,68 +1,72 @@
 
 ![Lightly Logo](docs/logos/lightly_logo_crop.png)
 
-
 ![GitHub](https://img.shields.io/github/license/lightly-ai/lightly)
 ![Unit Tests](https://github.com/lightly-ai/lightly/workflows/Unit%20Tests/badge.svg)
-![codecov](https://codecov.io/gh/lightly-ai/lightly/branch/master/graph/badge.svg?token=1NEAVROK3W)
+[![PyPI](https://img.shields.io/pypi/v/lightly)](https://pypi.org/project/lightly/)
+[![Downloads](https://pepy.tech/badge/lightly)](https://pepy.tech/project/lightly)
+[![Code style: black](https://img.shields.io/badge/code%20style-black-000000.svg)](https://github.com/psf/black)
```
Comment (philippmwirth) [inline README.md:8]: nice 😍

File: README.md [hunk 2]
```diff
@@ -77,264 +81,298 @@ Community and partner projects:
...
-# create a dataset from your image folder
-dataset = data.LightlyDataset(input_dir='./my/cute/cats/dataset/')
-
-# build a PyTorch dataloader
-dataloader = torch.utils.data.DataLoader(
-    dataset,                # pass the dataset to the dataloader
-    batch_size=128,         # a large batch size helps with the learning
-    shuffle=True,           # shuffling is important!
-    collate_fn=collate_fn)  # apply transformations to the input images
+from lightly import loss
```
Comment (philippmwirth) [inline README.md:123]: Did you run black on this?

File: README.md [hunk 3]
```diff
@@ -77,264 +81,298 @@ Community and partner projects:
...
 ```python
-trainer = pl.Trainer(max_epochs=max_epochs, devices=1, accelerator="gpu")
-trainer.fit(
-    model,
-    dataloader
-)
+from pytorch_lightning import Trainer
+
+trainer = Trainer(max_epochs=10, devices=1, accelerator="gpu")
+trainer.fit(model, dataloader)
```
Comment (philippmwirth) [inline README.md:225]: This is a bit misleading because the model above cannot be plugged into the lightning trainer.

File: README.md [hunk 4]
```diff
@@ -77,264 +81,298 @@ Community and partner projects:
...
-| SwaV          |        512 |    800 |             0.884 |
-| SMoG          |        512 |    800 |             0.800 |
+We provide multi-GPU training examples with distributed gather and synchronized BatchNorm.
+[Have a look at our docs regarding distributed training.](https://docs.lightly.ai/self-supervised-learning/getting_started/distributed_training.html)
+
+
+## Benchmarks
+
+Implemented models and their performance on various datasets. Hyperparameters are not
+tuned for maximum accuracy. Learn more about the benchmarks [here](https://docs.lightly.ai/self-supervised-learning/getting_started/benchmarks.html).
```
Comment (philippmwirth) [inline README.md:254]: Should we add here that imagenet benchmarks are coming soon?

File: README.md [hunk 5]
```diff
@@ -77,264 +81,298 @@ Community and partner projects:
...
+| SMoG          |        512 |    800 |                 0.800 |
+
 
 ## Terminology
 
-Below you can see a schematic overview of the different concepts present in the lightly Python package. The terms in bold are explained in more detail in our [documentation](https://docs.lightly.ai/self-supervised-learning/).
+Below you can see a schematic overview of the different concepts in the package.
+The terms in bold are explained in more detail in our [documentation](https://docs.lightly.ai/self-supervised-learning/).
 
 <img src="/docs/source/getting_started/images/lightly_overview.png" alt="Overview of the lightly pip package"/></a>
```
Comment (philippmwirth) [inline README.md:306]: This image is outdated :/ let's make a new one!

File: README.md [hunk 6]
```diff
@@ -77,264 +81,298 @@ Community and partner projects:
...
 - How can I contribute?
   - Create an issue if you encounter bugs or have ideas for features we should implement. You can also add your own code by forking this repository and creating a PR. More details about how to contribute with code is in our [contribution guide](CONTRIBUTING.md).
 
 - Is this framework for free?
-  - Yes, this framework is completely free to use and we provide the source code. We believe that
-  we need to make training deep learning models more data efficient to achieve widespread adoption. One step to achieve this goal is by leveraging self-supervised learning. The company behind Lightly is committed to keep this framework open-source.
+  - Yes, this framework is completely free to use and we provide the source code. We believe that we need to make training deep learning models more data efficient to achieve widespread adoption. One step to achieve this goal is by leveraging self-supervised learning. The company behind Lightly is committed to keep this framework open-source.
 
 - If this framework is free, how is the company behind lightly making money?
   - Training self-supervised models is only one part of our solution.
```
Comment (philippmwirth) [inline README.md:378]: Is the "Lightly in Research" section up-to-date? We should do a quick Google scholar search.

**Review by philippmwirth — CHANGES_REQUESTED** [review]
This looks already much better than before! Thanks a lot :) 

There's something we should discuss: We once decided to call the pip package "lightly" and the Lightly Solution "Lightly". If we change this now we should make sure it's a conscious decision.

File: README.md (hunk 2 above)
Comment (guarin) [reply README.md:123]: Yes :)

File: README.md (hunk 3 above)
Comment (guarin) [reply README.md:225]: I added an example model for pytorch lightning as well.

File: README.md (hunk 4 above)
Comment (guarin) [reply README.md:254]: Let's just add them once they are here :)

File: README.md (hunk 5 above)
Comment (guarin) [reply README.md:306]: Yes, we should also update it in the docs. Do you have the source?

Comment (guarin) [reply README.md]: Updated the image :) The only major change is that I added a `Transforms` field. Let me know what you think.

**Review by philippmwirth — APPROVED** [review]
Awesome!
